packages/cache/src/lib/CacheMiddleware.ts

```diff
@@ -0,0 +1,727 @@
+
+        return hook(data, key);
+      }
+    };
+
+    const { errors } = await cache[Method.Each]({ method: Method.Each, hook: cacheHook, errors: [] });
+
+    payload.metadata ??= {};
+    payload.metadata.skipProvider = true;
+    payload.errors = [...payload.errors, ...errors];
+
+    return payload;
+  }
+
+  @PreProvider()
+  public async [Method.Has](payload: Payloads.Has): Promise<Payloads.Has> {
+    const { key, path } = payload;
+    const { provider: cache } = this.context;
+    const getPayload = await cache[Method.Get]({ method: Method.Get, key, path, errors: [] });
+
+    if (isPayloadWithData<CacheMiddleware.Document<StoredValue>>(getPayload) && (await this.checkNotExpired(getPayload.data, key))) {
+      const { data: hasData, errors } = await cache[Method.Has]({ method: Method.Has, key, path: ['value', ...path], errors: [] });
+
+      payload.data = hasData;
+      payload.errors = [...payload.errors, ...errors];
+    }
+
+    return payload;
+  }
+
+  @PreProvider()
+  public async [Method.Entries](payload: Payloads.Entries<StoredValue>): Promise<Payloads.Entries<StoredValue>> {
+    const { provider: cache } = this.context;
+    const entries = await cache[Method.Entries]({ method: Method.Entries, errors: [] });
+
+    payload.errors = [...payload.errors, ...entries.errors];
+
+    if (isPayloadWithData<CacheMiddleware.Document<StoredValue>>(entries)) {
+      const { data } = entries;
+
+      payload.data ??= {};
+
+      for (const [key, val] of Object.entries(data)) {
+        if (await this.checkNotExpired(val, key)) {
+          payload.data[key] = val.value;
+        } else {
+          const bypassPayload = await this.provider[Method.Get]({ method: Method.Get, key, path: [], errors: [] });
+
+          if (isPayloadWithData<StoredValue>(bypassPayload)) {
+            const { data } = bypassPayload;
+
+            payload.data[key] = data;
+          }
+        }
+      }
+    }
+
+    return payload;
+  }
+
+  public async [Method.Every](payload: Payloads.Every.ByHook<StoredValue>): Promise<Payloads.Every.ByHook<StoredValue>>;
+  public async [Method.Every](payload: Payloads.Every.ByValue): Promise<Payloads.Every.ByValue>;
+
+  @PreProvider()
+  public async [Method.Every](payload: Payloads.Every<StoredValue>): Promise<Payloads.Every<StoredValue>> {
+    const { provider: cache } = this.context;
+    const { type, value } = payload;
+    const cacheHook = (hook: Payload.Hook<StoredValue, unknown>) => {
+      return async (value: CacheMiddleware.Document<StoredValue>, key: string) => {
+        if (await this.checkNotExpired(value, key)) {
+          return hook(value.value, key);
+        }
+
+        const bypassPayload = await this.provider[Method.Get]({ method: Method.Get, key, path: [], errors: [] });
+
+        if (isPayloadWithData<StoredValue>(bypassPayload)) {
+          const { data } = bypassPayload;
+
+          return hook(data, key);
+        }
+
+        return true; // I think(?) this is ok
```

Comment:
```suggestion
        return true;
```
Yes, the default of `Method.Every` is `true`